### PR TITLE
Feature/8396 encrypt event check in/8815 select correct downloading

### PR DESCRIPTION
**Pull request by KaiTeuber**
## Description
all downloads of TraceWarnings will now react on the app feature .unencryptedCheckinsEnabled
App Feature now can be decorated to make different behaviors much easier to realize.
This made it possible to add it to the debug menu very easy. Some more refactoring to the DeviceTime check was needed.
A bunch of Unittests have be updated to use the decorated app features - all ran green locally. 

## Link to Jira
https://jira-ibs.wbs.net.sap/browse/EXPOSUREAPP-8815

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift [hunk 1]
```diff
@@ -157,6 +158,7 @@ final class RiskProvider: RiskProviding {
 
 	private let store: Store
 	private let appConfigurationProvider: AppConfigurationProviding
+	private let appFeatureProvider: AppFeatureProviding
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift:161]: This is only used for the DeviceTimeCheck right? 
Why not store the DeviceTimeCheck directly during init?

**Review by marcussc — APPROVED** [review]
Small thing that good be improved otherwise LGTM

File: src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift (hunk 1 above)
Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift:161]: Good point, I'll try it.

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Services/Risk/Provider/RiskProvider.swift]: I did some more refactoring to get rid of a dependency cycle, please have another look

**Review by ArturFriesen — APPROVED** [review]

**Pull request merged**